patrickf949/SendIT-React
# #166269857: Enable users register to SendIT

### What does this PR do?
Enables users register to Send IT

#### Description of Task to be completed?
- Add routes for registration
- Add actions connecting to the backend
- Add reducers to handle registration action
- Add components for registration taking in username, email, password, contact
- Write tests for registration

#### How should this be manually tested?
Clone this repository

- `$ git clone git@github.com:patrickf949/SendIT-React.git`

Check into this branch

- `$ git checkout ft-register-166269857`

Install all dependencies

- `$ npm install`

Run the server

-  `$ npm start`

In your browser, on the SendIT interface, click the Sign Up button and enter your username, email, password and contact in order to successfully be registered onto the application.

#### What are the relevant pivotal tracker stories?
[#166269857](https://www.pivotaltracker.com/story/show/166269857)

#### Screenshots (if appropriate)

Sign up page

<img width="1440" alt="Screenshot 2019-05-30 at 17 35 17" src="https://user-images.githubusercontent.com/43245704/58640282-65584380-8301-11e9-9887-189dc19a1852.png">

Toast on successful sign up

<img width="1440" alt="Screenshot 2019-05-30 at 17 37 47" src="https://user-images.githubusercontent.com/43245704/58640438-bbc58200-8301-11e9-9d90-01f962ccaaa4.png">
